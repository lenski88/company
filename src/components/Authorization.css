.Authorization {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  border: 0.2em solid  #fbfc55;
  padding: 3em;
  margin-top:25vh
}

.Authorization .inputText {
    border-color: #fbfc55;
    background-color: #fbfc55 ;
    color:#0301a8;
    font-family:  'Press Start 2P', cursive;
    padding: 0.3em;
}
