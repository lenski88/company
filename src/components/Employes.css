.WrapperEmployes {
  display:flex;
  background-color: #0301a8 ;
  flex-flow: column wrap;
  align-items: center;
  position:absolute;
  top: 37%;
  width:100%;
  z-index: 1;
}

.NoWrapperEmployes {
display:none
}

.WrapperEmployes .ListEmployes {
  color: #fbfc55;
  text-align: center;
  font-size: 1.5em;
  margin: 0.5em;
  border: 0.1em solid;
  padding: 0.3em;
  box-shadow:  0.4em 0.4em  #b2b2b8e8;
}

.button {
  -webkit-transition-duration: 0.4s; /* Safari */
  transition-duration: 0.4s;
  margin: 1em;
  color: #0301a8;
  background-color: #fbfc55;
  border: "none";
  margin:0.9em;
  padding:0.9em;
  font-weight: bold;
  font-family: 'Press Start 2P', cursive;
}

.button:hover {
  color: #fbfc55;
  background-color: #0301a8;
}

@media (min-width: 1051px) {
  .WrapperEmployes {top: 30%;}
}

@media (max-width: 616px) {
  .WrapperEmployes {top: 50%;}
  .WrapperEmployes .ListEmployes{width: 100%;word-break: break-all;font-size: 1em;}
}

@media (max-width: 376px) {
  .WrapperEmployes {top: 50%;}
  .WrapperEmployes .ListEmployes{width: 100%;word-break: break-all;font-size: 0.6em;}
}

@media (max-height: 400px) {
  .WrapperEmployes {top: 55%;word-break: break-all;font-size: 0.6em}
}

