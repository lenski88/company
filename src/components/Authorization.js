import React from "react";
import PropTypes from "prop-types";

import "./Authorization.css";

class Authorization extends React.PureComponent {
  static propTypes = {
    employes: PropTypes.array.isRequired /* массив сотрудников */,
    login:
      PropTypes.bool.isRequired /* прошла ли авторизация; по умолчанию false */,
    cbGetUser: PropTypes.func,
  };

  state = {
    login: this.props.login,
    nameUser: "",
    errName: "",
    button: false,
  };

  nameUser = (eo) => {
    this.setState({
      nameUser: eo.target.value,
    });

    let arrayNames = this.props.employes;
    let name = eo.target.value;
    let isLogin = arrayNames.some((i) => {
      return name === i.login;
    });

    if (!isLogin) {
      this.setState({
        errName: "Нет такого сотрудника",
      });
    } else {
      this.setState({
        errName: "",
      });
    }
  };

  getUser = () => {
    if (!this.state.errName && !this.state.errPassword) {
      let arrayNames = this.props.employes;
      let name = this.state.nameUser;
      let user = arrayNames.find((i) => {
        return name === i.login;
      });
      this.props.cbGetUser(user);
      this.setState({
        nameUser: "",
        login: this.props.login,
      });
    }
  };

  render() {
    return (
      !this.state.login && (
        <div className={!this.props.login?"Authorization": "NoAuthorization"}>
          <label>Введите логин:</label>
          <br />
          <input
            className="inputText"
            type="text"
            name="nameUser"
            value={this.state.nameUser}
            onChange={this.nameUser}
            autoFocus
          ></input>
          <br />
          <span>{this.state.errName}</span>
          <br />
          <br />
          {this.state.nameUser && (
            <input
              className="button"
              type="button"
              onPointerDown={this.getUser}
              value="[Войти]"
              disabled={this.state.button}
            ></input>
          )}
        </div>
      )
    );
  }
}

export default Authorization;
