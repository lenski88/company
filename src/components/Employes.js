import React from "react";
import PropTypes from "prop-types";
import { TransitionGroup, CSSTransition } from "react-transition-group";

import SendTask from "./SendTask";
import ChangeEmployee from "./ChangeEmployee";

import "./Employes.css";

class Employes extends React.PureComponent {
  static propTypes = {
    employes: PropTypes.arrayOf(
      /* массив сотрудников */
      PropTypes.shape({
        id: PropTypes.number.isRequired, // id сотрудника
        login: PropTypes.string.isRequired, // логин для входа
        status: PropTypes.number.isRequired, // 0 - offline, 1 - online
        level: PropTypes.number.isRequired, // уровень доступа 1-низший, 2 - для начальников отделов, 3 - для руководства
        name: PropTypes.string.isRequired, // ФИО
        email: PropTypes.string.isRequired, // почта
        phone: PropTypes.string.isRequired, // номер телефона
        department: PropTypes.string.isRequired, // отдел
        position: PropTypes.string.isRequired, // должность
        task: PropTypes.array.isRequired, // массив задач
      })
    ),
    login: PropTypes.bool.isRequired,
    filterEmployes: PropTypes.array.isRequired,
    workMode: PropTypes.number.isRequired,
    isOpenWindowInHeader: PropTypes.number,
    cbNewTask: PropTypes.func,
    cbModeEmployes: PropTypes.func,
    cbChangeEmp: PropTypes.func,
    cbDeleteEmployee: PropTypes.func,
  };

  state = {
    employes: this.props.employes,
    employeeInfo: [],
    infoWindow: 0,
    mode: 0, // 1- добавить задачу, 2 - изменить пользователя
    idEmp: null, //id сотрудника, которому будет поставлена задача
    indexEmp: null, //индекс сотрудника в массиве
  };

  componentDidUpdate(oldProps) {
    if (this.props.login !== oldProps.login) {
      this.setState({
        infoWindow: 0,
        mode: 0,
      });
    }
    this.props.cbModeEmployes(this.state.mode);
  }

  info = (eo) => {
    let employes = this.props.employes;
    let employeeInfo;
    employeeInfo = employes.filter((i) => {
      return i.id === Number(eo.target.name);
    });
    this.setState({
      employeeInfo: employeeInfo,
      infoWindow: 1,
    });
  };

  exitInfo = () => {
    this.setState({
      infoWindow: 0,
    });
  };

  sendTask = (eo) => {
    this.setState({
      mode: 1,
      idEmp: Number(eo.target.name),
    });
  };

  newTask = (emp) => {
    this.setState({
      mode: 0,
      idEmp: null,
    });

    this.props.cbNewTask(emp);
  };

  exitSendTask = () => {
    this.setState({
      mode: 0,
      idEmp: null,
    });
  };

  changeEmployee = (eo) => {
    let employes = this.props.employes;
    let idEmp = Number(eo.target.name);
    let index = employes.findIndex((i) => {
      return i.id === idEmp;
    });
    this.setState({
      mode: 2,
      idEmp: idEmp,
      indexEmp: index,
    });
  };

  addChangeEmp = (emp) => {
    this.setState({
      mode: 0,
    });
    this.props.cbModeEmployes(this.state.mode);
    this.props.cbChangeEmp(emp);
  };

  changeExit = () => {
    this.setState({
      mode: 0,
    });
    this.props.cbModeEmployes(this.state.mode);
  };

  deleteEmployee = (eo) => {
    let deleteUser = this.props.employes.find((i) => {
      return i.id === Number(eo.target.name);
    });
    if (deleteUser.status === 1) {
      alert("Вы не можете удалить сотрудника, который сейчас в сети");
    } else {
      let isDelete = window.confirm("Удалить сотрудника?");
      if (isDelete) {
        this.props.cbDeleteEmployee(Number(eo.target.name));
        alert("Сотрудник удален");
      }
    }
  };

  render() {
    let employes; //список сотрудников
    let employeeInfo; // содержит хеш с инфо о сотруднике
    if (this.props.user) {
      let user = this.props.user;
      let employesList; // какой список сотрудников будет отрендерен: workMode = 1 - все, workMоde = 2 - отфильтрованный
      this.props.workMode === 1
        ? (employesList = this.props.employes)
        : (employesList = this.props.filterEmployes);

      employes = employesList.map((emp) => {
        if (emp.id !== user.id) {
          return (
            <CSSTransition
              classNames="list"
              key={emp.id}
              timeout={{ enter: 300, exit: 300 }}
            >
              <div className="ListEmployes" id={emp.id} key={emp.id}>
                {this.props.workMode === 1 && (
                  <div className={emp.status ? "online" : "offline"}></div>
                )}
                <p id="name">{emp.name}</p>
                <p id="position">{emp.position}</p>
                <br />
                <input
                  className="button"
                  type="button"
                  name={emp.id}
                  value="[Информация]"
                  onPointerDown={this.info}
                ></input>
                {user.level > 1 && (
                  <input
                    className="button"
                    type="button"
                    name={emp.id}
                    value="[Поставить задачу]"
                    onPointerDown={this.sendTask}
                  ></input>
                )}
                {user.level === 3 && (
                  <input
                    className="button"
                    type="button"
                    name={emp.id}
                    value="[Изменить]"
                    onPointerDown={this.changeEmployee}
                  ></input>
                )}
                {user.level === 3 && (
                  <input
                    className="button"
                    type="button"
                    name={emp.id}
                    value="[Удалить]"
                    onPointerDown={this.deleteEmployee}
                  ></input>
                )}
              </div>
            </CSSTransition>
          );
        }
      });

      employeeInfo = this.state.employeeInfo.map((i) => {
        return (
          <CSSTransition
            classNames="info"
            key={i.id}
            timeout={{ enter: 300, exit: 300 }}
          >
            <div className="ListEmployes" key={i.id}>
              <h2>Информация о сотруднике</h2>
              <p>ФИО:{i.name}</p>
              <p>Уровень доступа:{i.level}</p>
              <p>Почта:{i.email}</p>
              <p>Телефон:{i.phone}</p>
              <p>Отдел:{i.department}</p>
              <p>Дoлжность:{i.position}</p>
              <input
                className="button"
                type="button"
                value="[Вернуться к списку]"
                onPointerDown={this.exitInfo}
              ></input>
            </div>
          </CSSTransition>
        );
      });
    }

    return (
      this.props.login && (
        <React.Fragment>
          {this.state.mode === 0 && (
            <TransitionGroup
              component="div"
              className={
                this.props.isOpenWindowInHeader
                  ? "WrapperEmployes"
                  : "NoWrapperEmployes"
              }
            >
              {!this.state.infoWindow ? employes : employeeInfo}
            </TransitionGroup>
          )}
          {this.state.mode === 1 && (
            <SendTask
              employes={this.props.employes}
              user={this.props.user}
              idEmp={this.state.idEmp}
              cbNewTask={this.newTask}
              cbExitSendTask={this.exitSendTask}
            />
          )}
          {this.state.mode === 2 && (
            <ChangeEmployee
              employes={this.props.employes}
              idEmp={this.state.idEmp}
              indexEmp={this.state.indexEmp}
              cbAddChangeEmp={this.addChangeEmp}
              cbChangeExit={this.changeExit}
            />
          )}
        </React.Fragment>
      )
    );
  }
}

export default Employes;
