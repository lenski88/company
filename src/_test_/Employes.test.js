"use strict";

import React from "react";
import renderer from "react-test-renderer";
import { employes } from "./Authorization.test";
import { filterEmployes } from "./Authorization.test";
import { user } from "./Authorization.test";

import Employes from "../components/Employes";

let props = {
  employes: employes,
  user: user,
  filterEmployes: filterEmployes,
  workMode: 1,
  login: true,
};


test("проверка правильности отображения данных о сотруднике", () => {
  props = { ...props, workMode: 2 };
  let component = renderer.create(<Employes {...props} />);

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();

  let emp = component.root.find((el) => el.props.id == 39);

  let name = emp.find((el) => el.props.id == "name");
  expect(name.children.join("")).toBe("Михалова Розалина Геласьевна");

  let position = emp.find((el) => el.props.id == "position");
  expect(position.children.join("")).toBe(
    "Специалист по кадровому делопроизводству"
  );
});


test("проверка employes", () => {
  let component = renderer.create(<Employes {...props} />);

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();

  props = { ...props, login: false };
  component = renderer.create(<Employes {...props} />);

  componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});
