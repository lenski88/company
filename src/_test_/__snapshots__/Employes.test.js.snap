// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка Authorization 1`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <span>
    
  </span>
  <br />
  <br />
  
</div>
`;

exports[`проверка Authorization 2`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="sdfasdf"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка Authorization 3`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка Authorization 4`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="saveliev47"
  />
  <br />
  <span>
    
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка employes 1`] = `
<div
  className="NoWrapperEmployes"
>
  <div
    className="ListEmployes"
    id={35}
  >
    <p
      id="name"
    >
      Смирнова Августа Мироновна
    </p>
    <p
      id="position"
    >
      Начальник отдела
    </p>
    <br />
    <input
      className="button"
      name={35}
      onPointerDown={[Function]}
      type="button"
      value="[Информация]"
    />
    <input
      className="button"
      name={35}
      onPointerDown={[Function]}
      type="button"
      value="[Поставить задачу]"
    />
    <input
      className="button"
      name={35}
      onPointerDown={[Function]}
      type="button"
      value="[Изменить]"
    />
    <input
      className="button"
      name={35}
      onPointerDown={[Function]}
      type="button"
      value="[Удалить]"
    />
  </div>
  <div
    className="ListEmployes"
    id={36}
  >
    <p
      id="name"
    >
      Волкова Альжбета Наумовна
    </p>
    <p
      id="position"
    >
      Специалист по кадровому делопроизводству
    </p>
    <br />
    <input
      className="button"
      name={36}
      onPointerDown={[Function]}
      type="button"
      value="[Информация]"
    />
    <input
      className="button"
      name={36}
      onPointerDown={[Function]}
      type="button"
      value="[Поставить задачу]"
    />
    <input
      className="button"
      name={36}
      onPointerDown={[Function]}
      type="button"
      value="[Изменить]"
    />
    <input
      className="button"
      name={36}
      onPointerDown={[Function]}
      type="button"
      value="[Удалить]"
    />
  </div>
  <div
    className="ListEmployes"
    id={37}
  >
    <p
      id="name"
    >
      Шестакова Марьяна Рудольфовна
    </p>
    <p
      id="position"
    >
      Специалист по кадровому делопроизводству
    </p>
    <br />
    <input
      className="button"
      name={37}
      onPointerDown={[Function]}
      type="button"
      value="[Информация]"
    />
    <input
      className="button"
      name={37}
      onPointerDown={[Function]}
      type="button"
      value="[Поставить задачу]"
    />
    <input
      className="button"
      name={37}
      onPointerDown={[Function]}
      type="button"
      value="[Изменить]"
    />
    <input
      className="button"
      name={37}
      onPointerDown={[Function]}
      type="button"
      value="[Удалить]"
    />
  </div>
  <div
    className="ListEmployes"
    id={38}
  >
    <p
      id="name"
    >
      Горбачёва Харита Георгиевна
    </p>
    <p
      id="position"
    >
      Специалист по кадровому делопроизводству
    </p>
    <br />
    <input
      className="button"
      name={38}
      onPointerDown={[Function]}
      type="button"
      value="[Информация]"
    />
    <input
      className="button"
      name={38}
      onPointerDown={[Function]}
      type="button"
      value="[Поставить задачу]"
    />
    <input
      className="button"
      name={38}
      onPointerDown={[Function]}
      type="button"
      value="[Изменить]"
    />
    <input
      className="button"
      name={38}
      onPointerDown={[Function]}
      type="button"
      value="[Удалить]"
    />
  </div>
  <div
    className="ListEmployes"
    id={39}
  >
    <p
      id="name"
    >
      Михалова Розалина Геласьевна
    </p>
    <p
      id="position"
    >
      Специалист по кадровому делопроизводству
    </p>
    <br />
    <input
      className="button"
      name={39}
      onPointerDown={[Function]}
      type="button"
      value="[Информация]"
    />
    <input
      className="button"
      name={39}
      onPointerDown={[Function]}
      type="button"
      value="[Поставить задачу]"
    />
    <input
      className="button"
      name={39}
      onPointerDown={[Function]}
      type="button"
      value="[Изменить]"
    />
    <input
      className="button"
      name={39}
      onPointerDown={[Function]}
      type="button"
      value="[Удалить]"
    />
  </div>
</div>
`;

exports[`проверка employes 2`] = `null`;

exports[`проверка нажатия на кнопку войти 1`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <span>
    
  </span>
  <br />
  <br />
  
</div>
`;

exports[`проверка нажатия на кнопку войти 2`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка нажатия на кнопку войти 3`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка нажатия на кнопку войти 4`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;

exports[`проверка правильности отображения данных о сотруднике 1`] = `
<div
  className="NoWrapperEmployes"
>
  <div
    className="ListEmployes"
    id={35}
  >
    <p
      id="name"
    >
      Смирнова Августа Мироновна
    </p>
    <p
      id="position"
    >
      Начальник отдела
    </p>
    <br />
    <input
      className="button"
      name={35}
      onPointerDown={[Function]}
      type="button"
      value="[Информация]"
    />
    <input
      className="button"
      name={35}
      onPointerDown={[Function]}
      type="button"
      value="[Поставить задачу]"
    />
    <input
      className="button"
      name={35}
      onPointerDown={[Function]}
      type="button"
      value="[Изменить]"
    />
    <input
      className="button"
      name={35}
      onPointerDown={[Function]}
      type="button"
      value="[Удалить]"
    />
  </div>
  <div
    className="ListEmployes"
    id={36}
  >
    <p
      id="name"
    >
      Волкова Альжбета Наумовна
    </p>
    <p
      id="position"
    >
      Специалист по кадровому делопроизводству
    </p>
    <br />
    <input
      className="button"
      name={36}
      onPointerDown={[Function]}
      type="button"
      value="[Информация]"
    />
    <input
      className="button"
      name={36}
      onPointerDown={[Function]}
      type="button"
      value="[Поставить задачу]"
    />
    <input
      className="button"
      name={36}
      onPointerDown={[Function]}
      type="button"
      value="[Изменить]"
    />
    <input
      className="button"
      name={36}
      onPointerDown={[Function]}
      type="button"
      value="[Удалить]"
    />
  </div>
  <div
    className="ListEmployes"
    id={37}
  >
    <p
      id="name"
    >
      Шестакова Марьяна Рудольфовна
    </p>
    <p
      id="position"
    >
      Специалист по кадровому делопроизводству
    </p>
    <br />
    <input
      className="button"
      name={37}
      onPointerDown={[Function]}
      type="button"
      value="[Информация]"
    />
    <input
      className="button"
      name={37}
      onPointerDown={[Function]}
      type="button"
      value="[Поставить задачу]"
    />
    <input
      className="button"
      name={37}
      onPointerDown={[Function]}
      type="button"
      value="[Изменить]"
    />
    <input
      className="button"
      name={37}
      onPointerDown={[Function]}
      type="button"
      value="[Удалить]"
    />
  </div>
  <div
    className="ListEmployes"
    id={38}
  >
    <p
      id="name"
    >
      Горбачёва Харита Георгиевна
    </p>
    <p
      id="position"
    >
      Специалист по кадровому делопроизводству
    </p>
    <br />
    <input
      className="button"
      name={38}
      onPointerDown={[Function]}
      type="button"
      value="[Информация]"
    />
    <input
      className="button"
      name={38}
      onPointerDown={[Function]}
      type="button"
      value="[Поставить задачу]"
    />
    <input
      className="button"
      name={38}
      onPointerDown={[Function]}
      type="button"
      value="[Изменить]"
    />
    <input
      className="button"
      name={38}
      onPointerDown={[Function]}
      type="button"
      value="[Удалить]"
    />
  </div>
  <div
    className="ListEmployes"
    id={39}
  >
    <p
      id="name"
    >
      Михалова Розалина Геласьевна
    </p>
    <p
      id="position"
    >
      Специалист по кадровому делопроизводству
    </p>
    <br />
    <input
      className="button"
      name={39}
      onPointerDown={[Function]}
      type="button"
      value="[Информация]"
    />
    <input
      className="button"
      name={39}
      onPointerDown={[Function]}
      type="button"
      value="[Поставить задачу]"
    />
    <input
      className="button"
      name={39}
      onPointerDown={[Function]}
      type="button"
      value="[Изменить]"
    />
    <input
      className="button"
      name={39}
      onPointerDown={[Function]}
      type="button"
      value="[Удалить]"
    />
  </div>
</div>
`;

exports[`проверка рендера кнопки при верном логине 1`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <span>
    
  </span>
  <br />
  <br />
  
</div>
`;

exports[`проверка рендера кнопки при верном логине 2`] = `
<div
  className="Authorization"
>
  <label>
    Введите логин:
  </label>
  <br />
  <input
    autoFocus={true}
    className="inputText"
    name="nameUser"
    onChange={[Function]}
    type="text"
    value="ivanov0"
  />
  <br />
  <span>
    Нет такого сотрудника
  </span>
  <br />
  <br />
  <input
    className="button"
    disabled={false}
    onPointerDown={[Function]}
    type="button"
    value="[Войти]"
  />
</div>
`;
